modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java

```diff
@@ -778,6 +779,49 @@ public void testAddObjectAction() throws Exception {
 		_objectActionLocalService.deleteObjectAction(systemObjectAction);
 	}
 
+	@Test
+	public void testAddObjectActionWithCircularReference() throws Exception {
+		_testAddObjectActionWithCircularReference(2);
+
+		Class<ObjectActionThreadLocal> clazz = ObjectActionThreadLocal.class;
+
+		String originalFieldName = "_clearObjectEntryIdsMapThreadLocal";
+
+		Object originalField = ReflectionTestUtil.getFieldValue(
+			clazz, originalFieldName);
+
+		int expectedObjectEntriesCount = RandomTestUtil.randomInt(3, 20);
+
+		try {
+			ThreadLocal<Boolean> trueThreadLocal = new ThreadLocal<Boolean>() {
+
+				@Override
+				public Boolean get() {
+					if (_count < expectedObjectEntriesCount) {
+						_count++;
+
+						return true;
+					}
+
+					return false;
+				}
+
+				private int _count = 1;
```

Comment:
@devpedrotavares can we rename it to `actualObjectEntriesCount`?

Reply:
I would just point out that this is not **exactly** the actual object entries count, because we start from 1 before we even add the first entry.